allure-cucumber5-jvm/src/main/java/io/qameta/allure/cucumber5jvm/AllureCucumber5Jvm.java

```diff
@@ -0,0 +1,415 @@
+                        dataTableCsv.append(rows.get(i));
+                    }
+                    else{
+                        dataTableCsv.append(rows.get(i)+"\t");
+                    }
+                }
+                dataTableCsv.append('\n');
+            }
+        }
+        final String attachmentSource = lifecycle
+                .prepareAttachment("Data table", "text/tab-separated-values", "csv");
+        lifecycle.writeAttachment(attachmentSource,
+                new ByteArrayInputStream(dataTableCsv.toString().getBytes(StandardCharsets.UTF_8)));
+    }
+
+    private void handleHookStep(final TestStepFinished event) {
+        final HookTestStep hookStep = (HookTestStep) event.getTestStep();
+        final String uuid = getHookStepUuid(hookStep);
+        final FixtureResult fixtureResult = new FixtureResult().setStatus(translateTestCaseStatus(event.getResult()));
+
+        if (!Status.PASSED.equals(fixtureResult.getStatus())) {
+            final TestResult testResult = new TestResult().setStatus(translateTestCaseStatus(event.getResult()));
+            final StatusDetails statusDetails = getStatusDetails(event.getResult().getError())
+                    .orElseGet(StatusDetails::new);
+
+            final String errorMessage = event.getResult().getError() == null ? hookStep.getHookType()
+                    .name() + " is failed." : hookStep.getHookType()
+                    .name() + " is failed: " + event.getResult().getError().getLocalizedMessage();
+            statusDetails.setMessage(errorMessage);
+
+            if (hookStep.getHookType() == HookType.BEFORE) {
+                final TagParser tagParser = new TagParser(currentFeature.get(), currentTestCase.get());
+                statusDetails
+                        .setFlaky(tagParser.isFlaky())
+                        .setMuted(tagParser.isMuted())
+                        .setKnown(tagParser.isKnown());
+                testResult.setStatus(Status.SKIPPED);
+                updateTestCaseStatus(testResult.getStatus());
+                forbidTestCaseStatusChange.set(true);
+            } else {
+                testResult.setStatus(Status.BROKEN);
+                updateTestCaseStatus(testResult.getStatus());
+            }
+            fixtureResult.setStatusDetails(statusDetails);
+        }
+
+        lifecycle.updateFixture(uuid, result -> result.setStatus(fixtureResult.getStatus())
+                .setStatusDetails(fixtureResult.getStatusDetails()));
+        lifecycle.stopFixture(uuid);
+    }
+
+    private void handlePickleStep(final TestStepFinished event) {
+
+        final Status stepStatus = translateTestCaseStatus(event.getResult());
+        final StatusDetails statusDetails;
+        if (event.getResult().getStatus() == io.cucumber.plugin.event.Status.UNDEFINED) {
+            updateTestCaseStatus(Status.PASSED);
+
+            statusDetails =
+                    getStatusDetails(new PendingException("TODO: implement me"))
```

Comment:
An undefined step means that Cucumber couldn't find the step. Using the pending exception here seems odd because pending means the step exists but wasn't implemented yet.

Reply:
The intention was to flag also unimplemented steps in report. I guess here also IllegalState would suffice.